src/Chapter3.hs

```diff
@@ -480,7 +532,81 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
-{-
+data Castle
+    = Castle String
+    | NoCastle
+
+data Wall
+    = Wall
+    | NoWall
+
+-- tried to give both Wall and Castle a None alternative but the compilare complained
+-- about multiple declarations of None, is there any way to bypass this ?
+
+data Center
+    = Church
+    | Library
+
+data House = MkHouse
+    { people  :: Int }
+
+data City = MkCity
+    { cityCastle  :: Castle
+    , cityWall    :: Wall
+    , cityCenter :: Center
+    , cityHouses  :: [House]
+    }
+
+
+countPeople :: [House] -> Int
+countPeople [] = 0
+countPeople [x] = (people x)
```

Comment:
This case is asctually covered by the next one.

Also, you can notice that it is actually `sum` function + standard `map` functions 🙂